framework/src/packets/ip/v6/srh.rs

```diff
@@ -491,6 +487,53 @@ impl<E: Ipv6Packet> IpPacket for SegmentRouting<E> {
 
 impl<E: Ipv6Packet> Ipv6Packet for SegmentRouting<E> {}
 
+impl<E: Ipv6Packet> SegmentsIndex for SegmentRouting<E> {
+    /// Returns the first segment in the segment list.
+    #[inline]
+    fn first_segment(&self) -> Segment {
+        let segments = self.segments();
+        segments[segments.len() - 1]
+    }
+
+    /// Returns the last segment in the segment list.
+    #[inline]
+    fn last_segment(&self) -> Segment {
+        self.segments()[0]
+    }
+
+    /// Returns the current segment in the segment list.
+    #[inline]
+    fn current_segment(&self) -> Segment {
+        self.segments()[self.segments_left() as usize]
+    }
+
+    /// Returns the previous segment in the segment list.
+    #[inline]
+    fn previous_segment(&self) -> Result<Segment> {
+        let segments = self.segments();
+        let segments_left = self.segments_left() as usize;
+        let prev_idx = segments_left as usize + 1;
+        if prev_idx < segments.len() {
```

Comment:
you can skip the manual bounds check by using `slice.get` and map the returned option to a result.

Reply:
nevermind, after a discussion this is not worth it.